File: LeetCode/1732.py
#Link: https://leetcode.com/problems/find-the-highest-altitude/

class Solution:
    def largestAltitude(self, gain: List[int]) -> int:
        curr = maxheight = 0
        for x in gain:
            curr += x
            maxheight = max(curr,maxheight)
        return maxheight
        

File: LeetCode/1732.cpp
//Link: https://leetcode.com/problems/find-the-highest-altitude/

class Solution {
public:
    int largestAltitude(vector<int>& gain) {
        int curr = 0, max = 0;
        for(int x : gain) {
            curr += x;
            max = (curr > max) ? curr : max;
        }
        return max;
    }
};
